tools/nuget/generate_nuspec_for_native_nuget.py

```diff
@@ -57,7 +60,11 @@ def generate_file_list_for_ep(nuget_artifacts_dir, ep, files_list, include_pdbs,
                 child = child / "lib"  # noqa: PLW2901
                 for child_file in child.iterdir():
                     suffixes = [".dll", ".lib", ".pdb"] if include_pdbs else [".dll", ".lib"]
-                    if child_file.suffix in suffixes and is_this_file_needed(ep, child_file.name):
+                    if (
+                        child_file.suffix in suffixes
+                        and is_this_file_needed(ep, child_file.name, package_name)
+                        and package_name != "Microsoft.ML.OnnxRuntime.Gpu-linux"
```

Comment:
It'd better to keep it as before.
Because we only split win/linux files for gpu packages.
It's clear that there's special changes for GPU package.
I would like to be defensive to avoid regression.

Reply:
Can we maybe move `is_this_file_needed` to define it inside of `generate_file_list_for_ep`? It has a very generic name but it's usage is specific to that function. Doing that will make it easier to see the multiple places where there is special logic for the GPU package.